### PR TITLE
jsonpb: properly support ListValue well known type

### PR DESCRIPTION
**Pull request by alanshiehclone**
Summary:
golang/protobuf/jsonpb does not serialize/deserialize ListValue according to the proto3
JSON spec. ListValue is supposed to be translated into a JSON array. Without this change,
ser-des outputs ListValue as a generic protobuf type, e.g.

ListValue{Values: {Value: Value_NumberValue{NumberValue: 1}}}
=> {"Values": [1]}

Correct output should be
=> [1]

### CODE REVIEW

File: protoc-gen-go/testdata/Makefile [hunk 1]
```diff
@@ -57,7 +57,7 @@ testbuild:	regenerate
 
 regenerate:
 	# Invoke protoc once to generate three independent .pb.go files in the same package.
-	protoc --go_out=. multi/multi{1,2,3}.proto
```
Comment (msolo-dropbox) [inline protoc-gen-go/testdata/Makefile:60]: Why change this?

Comment (alanshiehclone) [reply protoc-gen-go/testdata/Makefile:60]: Make didn't pick up the right shell on my dev VM, hence globbing failed. So I rewrote this to reduce the dependency on shell.

File: jsonpb/jsonpb_test.go [hunk 2]
```diff
@@ -397,6 +397,42 @@ var marshalingTests = []struct {
 	{"BoolValue", marshaler, &pb.KnownTypes{Bool: &wpb.BoolValue{Value: true}}, `{"bool":true}`},
 	{"StringValue", marshaler, &pb.KnownTypes{Str: &wpb.StringValue{Value: "plush"}}, `{"str":"plush"}`},
 	{"BytesValue", marshaler, &pb.KnownTypes{Bytes: &wpb.BytesValue{Value: []byte("wow")}}, `{"bytes":"d293"}`},
+	{"ListValue nil", marshaler, &pb.KnownTypes{Listval: nil}, `{}`},
```
Comment (msolo-dropbox) [inline jsonpb/jsonpb_test.go:400]: Looking at the repeated tests, I'm not sure if I understand the situation. Repeated fields are clearly supported, so what is the difference between "ListValue" and repeated field?

**Comment by alanshiehclone** [comment]
Closing this request for now -- was asked to hold off until I get OSS approval

**Pull request merged**